### PR TITLE
Added `eval` in sympy.functions.special.bessel.jn

### PR DESCRIPTION
**Pull request by czgdp1807**
#### References to other Issues or PRs

Closes #12116

#### Other comments
I was wondering if we can define `jn` in terms of Bessel Function of the first kind instead of some special inputs. Let me know your views.

#### Release Notes

* functions
  * `sympy.functions.special.bessel.jn` now has `eval` method

### CODE REVIEW

**Comment by czgdp1807** [comment]
ping @jksuom @asmeurer

**Comment by jksuom** [comment]
I would explicitly evaluate jn only for non-negative integer orders.

**Comment by czgdp1807** [comment]
@jksuom Can you please elaborate more? What acutally is needed when `nu.is_nonnegative` is `True`?

**Comment by jksuom** [comment]
`jn` is regular analytic at 0 only when nu is a non-negative integer n. I would assign a value for those orders only (1 for n = 0; 0 for n = 1, 2, ...).

**Comment by czgdp1807** [comment]
ping @jksuom I have made some changes. Please take a look and let me know of any further requirements.

**Comment by jksuom** [comment]
Looks good. Perhaps (limit) value 0 at +/-oo could also be added.

**Comment by czgdp1807** [comment]
ping @jksuom I have made some updates. Please take a look and merge it if the tests are passing if you think it's good. Thanks.

File: sympy/functions/special/bessel.py [hunk 1]
```diff
@@ -670,15 +670,16 @@ class jn(SphericalBesselBase):
...
         if z.is_zero:
-            if nu.is_zero: # jn(0,0)=1
+            if nu.is_zero:
                 return S.One
             elif nu.is_integer:
                 if nu.is_positive:
                     return S.Zero
                 else:
                     return S.ComplexInfinity
+        if z.is_infinite:
```
Comment (jksuom) [inline sympy/functions/special/bessel.py]: This should be oo or -oo. Not true for zoo, I*oo, etc.

**Comment by jksuom** [comment]
Thanks.

**Pull request merged**